api/transport/errors.go

```diff
@@ -63,3 +63,18 @@ func IsUnrecognizedProcedureError(err error) bool {
 	_, ok := err.(errors.UnrecognizedProcedureError)
 	return ok
 }
+
+// UnrecognizedEncodingError returns an error for the given request, such that
+// IsUnrecognizedEncodingError can distinguish it from other errors coming out
+// of router.Choose.
+func UnrecognizedEncodingError(req *Request, want []string) error {
+	return errors.RouterUnrecognizedEncodingError(want, string(req.Encoding))
+}
+
+// IsUnrecognizedEncodingError returns true for errors returned by
+// Router.Choose if the router cannot find a handler for the request's
+// encoding.
+func IsUnrecognizedEncodingError(err error) bool {
```

Comment:
Do we really need these? This is expanding on an API that we're about to do work on with the error stuff @kriskowal